internal/test/mock_clair.go

```diff
@@ -128,26 +135,53 @@ func (c *ClairDouble) postIndexReport(w http.ResponseWriter, r *http.Request) {
 }
 
 func (c *ClairDouble) getIndexReport(w http.ResponseWriter, r *http.Request) {
-	httpapi.SkipRequestLog(r)
 	httpapi.IdentifyEndpoint(r, "/indexer/api/v1/index_report/{digest}")
 
 	digest := mux.Vars(r)["digest"]
-	if c.WasIndexSubmitted[digest] {
-		state := "CheckManifest"
-		if c.ReportFixtures[digest] != "" {
-			state = "IndexFinished"
-		}
-		respondwith.JSON(w, http.StatusCreated, map[string]interface{}{
-			"manifest_hash": digest,
-			"state":         state,
-		})
-	} else {
+
+	if !c.WasIndexSubmitted[digest] {
 		http.Error(w, "not found", http.StatusNotFound)
+		return
 	}
+
+	indexReportFixture := c.IndexReportFixtures[digest]
+	if indexReportFixture != "" {
+		file, err := os.Open(indexReportFixture)
+		if respondwith.ErrorText(w, err) {
+			return
+		}
+		content, err := io.ReadAll(file)
+		if respondwith.ErrorText(w, err) {
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write(content)
+		return
+	}
+
+	state := "CheckManifest"
+	if c.ReportFixtures[digest] != "" {
+		state = "IndexFinished"
+	}
+	respondwith.JSON(w, http.StatusCreated, map[string]interface{}{
+		"manifest_hash": digest,
+		"state":         state,
+	})
+}
+
+func (c *ClairDouble) deleteIndexReport(w http.ResponseWriter, r *http.Request) {
```

Comment:
This function should include consistency checks similar to the other ClairDouble request handlers:

- Only expected manifests should be permitted, i.e. there should be an `c.IndexFixtures[digest]`.
- Only manifests that were submitted for indexing should be permitted, i.e. `c.WasIndexSubmitted[digest]` should be true.